src/funnels/Depositor.sol

```diff
@@ -0,0 +1,424 @@
+        external
+        payable
+        returns (uint256 amount0, uint256 amount1);
+
+    struct CollectParams {
+        uint256 tokenId;
+        address recipient;
+        uint128 amount0Max;
+        uint128 amount1Max;
+    }
+
+    /// @notice Collects up to a maximum amount of fees owed to a specific position to the recipient
+    /// @param params tokenId The ID of the NFT for which tokens are being collected,
+    /// recipient The account that should receive the tokens,
+    /// amount0Max The maximum amount of token0 to collect,
+    /// amount1Max The maximum amount of token1 to collect
+    /// @return amount0 The amount of fees collected in token0
+    /// @return amount1 The amount of fees collected in token1
+    function collect(CollectParams calldata params) external payable returns (uint256 amount0, uint256 amount1);
+
+}
+
+interface UniV3PoolLike {
+    function slot0() external view returns (
+        uint160 sqrtPriceX96,
+        int24 tick,
+        uint16 observationIndex,
+        uint16 observationCardinality,
+        uint16 observationCardinalityNext,
+        uint8 feeProtocol,
+        bool unlocked
+    );
+}
+
+contract Depositor {
+    mapping (address => uint256) public wards;
+    mapping (address => mapping (address => uint256)) public hops;       // [seconds]   hops[gem0][gem1] is the cooldown one has to wait between changes to the liquidity of a (gem0, gem1) pool
+    mapping (address => mapping (address => uint256)) public zzz;        // [seconds]    zzz[gem0][gem1] is the timestamp of the last liquidity change for a (gem0, gem1) pool
+    mapping (address => mapping (address => Cap)) public caps;           // [amount]    caps[gem0][gem1] is the tuple (amt0, amt1) indicating the maximum amount of (gem0, gem1) that can be added as liquidity each hop for a (gem0, gem1) pool
+
+    mapping (bytes32 => uint256) public tokenIds;  // key = keccak256(abi.encode(gem0, gem1, fee, tickLower, tickUpper)) => tokenId of the liquidity position
+
+    address   public immutable buffer;                // Contract from/to which the two tokens that make up the liquidity position are pulled/pushed
+    RolesLike public immutable roles;                 // Contract managing access control for this Depositor
+    bytes32   public immutable ilk;
+    address   public immutable uniV3Factory;          // 0x1F98431c8aD98523631AE4a59f267346ea31F984
+    address   public immutable uniV3PositionManager;  // 0xC36442b4a4522E871399CD717aBDD847Ab11FE88
+
+    struct Cap {
+        uint128 amt0;
+        uint128 amt1;
+    }
+
+    event Rely (address indexed usr);
+    event Deny (address indexed usr);
+    event File (bytes32 indexed what, address indexed gemA, address indexed gemB, uint256 data);
+    event File (bytes32 indexed what, address indexed gemA, address indexed gemB, uint128 dataA, uint128 dataB);
+    event File (bytes32 indexed what, address data);
```

Comment:
```suggestion
    event Rely(address indexed usr);
    event Deny(address indexed usr);
    event File(bytes32 indexed what, address indexed gemA, address indexed gemB, uint256 data);
    event File(bytes32 indexed what, address indexed gemA, address indexed gemB, uint128 dataA, uint128 dataB);
    event File(bytes32 indexed what, address data);
```

Reply:
Done in https://github.com/makerdao/dss-allocator/pull/15